### PR TITLE
Flags

### PR DESCRIPTION
**Pull request by AynuraKM**
Unused constants removed. 
Incompatibility conflicts with the DPDK api fixed.

### CODE REVIEW

File: ethdev/ethdev.go [hunk 1]
```diff
@@ -201,8 +105,6 @@ type RxMode struct {
 	// The multi-queue packet distribution mode to be used, e.g. RSS.
 	// See MqRx* constants.
 	MqMode uint
-	// Only used if JUMBO_FRAME enabled.
-	MaxRxPktLen uint32
```
Comment (yerden) [inline ethdev/ethdev.go:205]: This doesn't belong to this commit. Not a constant.

File: ethdev/flow/item_eth.go [hunk 2]
```diff
@@ -5,12 +5,23 @@ package flow
...
 
+enum {
+	ITEM_ETH_OFF_HDR = offsetof(struct rte_flow_item_eth, hdr),
+};
+
+#if RTE_VERSION < RTE_VERSION_NUM(21, 11, 0, 0)
 enum {
 	ETHER_HDR_OFF_SRC = offsetof(struct rte_ether_hdr, s_addr),
 	ETHER_HDR_OFF_DST = offsetof(struct rte_ether_hdr, d_addr),
-	ITEM_ETH_OFF_HDR = offsetof(struct rte_flow_item_eth, hdr),
```
Comment (yerden) [inline ethdev/flow/item_eth.go:12]: Not used? ok then

File: ethdev/flow/item_ipv4.go [hunk 3]
```diff
@@ -55,6 +60,11 @@ func cvtIPv4Header(dst *C.struct_rte_ipv4_hdr, src *layers.IPv4) {
 	}
 }
 
+func setVersion(dst *C.struct_rte_ipv4_hdr, src *layers.IPv4) {
```
Comment (yerden) [inline ethdev/flow/item_ipv4.go:63]: can you come up this less ambiguous name of the function? `setVersion` sounds too general.

File: ethdev/ethdev.go [hunk 4]
```diff
@@ -14,6 +14,17 @@ package ethdev
 #include <rte_errno.h>
 #include <rte_memory.h>
 #include <rte_ethdev.h>
+#include <rte_version.h>
+
+#if RTE_VERSION < RTE_VERSION_NUM(21, 11, 0, 0)
```
Comment (yerden) [inline ethdev/ethdev.go:19]: you sure about 21.11? it'd be nice to put exact commit id as a comment here for descendants to behold...

File: ethdev/ethdev.go [hunk 5]
```diff
@@ -350,9 +363,16 @@ func OptRxMode(conf RxMode) Option {
 			split_hdr_size: C.ushort(conf.SplitHdrSize),
 			offloads:       C.ulong(conf.Offloads),
 		}
+		setRxPktLen(c, conf.MTU)
 	}}
 }
 
+func setRxPktLen(c *ethConf, n uint32) {
```
Comment (yerden) [inline ethdev/ethdev.go:370]: could be method of `ethConf`?

File: ethdev/ethdev.go [hunk 6]
```diff
@@ -350,9 +363,16 @@ func OptRxMode(conf RxMode) Option {
...
+	rxptr := unsafe.Pointer((*C.struct_rte_eth_rxmode)(&c.conf.rxmode))
+	p := unsafe.Pointer(uintptr(rxptr) + C.RX_MODE_LEN_OFF)
+	*(*C.uint)(p) = (C.uint)(n)
```
Comment (yerden) [inline ethdev/ethdev.go:373]: Well that's risky. Using `uint32` and `C.uint32_t` is interchangeable and more portable than `C.uint`.

File: ethdev/flow/item_eth.go (hunk 2 above)
Comment (AynuraKM) [reply ethdev/flow/item_eth.go:12]: Эта строчка выше находится
